### PR TITLE
#114 skip pre-release versions when picking the last minor release

### LINKED ISSUE #114: Undefined offset: 0 in ... PickLastMinorVersionFromCollection
I add the the BCchecker locally to our project at https://github.com/dmstr/yii2-adminlte-asset

When I run the check in the container I get:

```
root@f5eb6f0ecb4d:/repo# vendor/bin/roave-backward-compatibility-check 
PHP Notice:  Undefined offset: 0 in /repo/vendor/roave/backward-compatibility-check/src/Git/PickLastMinorVersionFromCollection.php on line 48

Notice: Undefined offset: 0 in /repo/vendor/roave/backward-compatibility-check/src/Git/PickLastMinorVersionFromCollection.php on line 48
PHP Fatal error:  Uncaught TypeError: Return value of Roave\BackwardCompatibility\Git\PickLastMinorVersionFromCollection::forVersions() must be an instance of Version\Version, null returned in /repo/vendor/roave/backward-compatibility-check/src/Git/PickLastMinorVersionFromCollection.php:48
Stack trace:
#0 /repo/vendor/roave/backward-compatibility-check/src/Command/AssertBackwardsCompatible.php(226): Roave\BackwardCompatibility\Git\PickLastMinorVersionFromCollection->forVersions(Object(Version\VersionsCollection))
#1 /repo/vendor/roave/backward-compatibility-check/src/Command/AssertBackwardsCompatible.php(150): Roave\BackwardCompatibility\Command\AssertBackwardsCompatible->determineFromRevisionFromRepository(Object(Roave\BackwardCompatibility\Git\CheckedOutRepository), Object(Symfony\Component\Console\Output\StreamOutput))
#2 /repo/vendor/symfony/console/Command/Command.php(255): Roave\BackwardCompatibility\Command\AssertBackwardsCompatible->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Con in /repo/vendor/roave/backward-compatibility-check/src/Git/PickLastMinorVersionFromCollection.php on line 48

Fatal error: Uncaught TypeError: Return value of Roave\BackwardCompatibility\Git\PickLastMinorVersionFromCollection::forVersions() must be an instance of Version\Version, null returned in /repo/vendor/roave/backward-compatibility-check/src/Git/PickLastMinorVersionFromCollection.php:48
Stack trace:
#0 /repo/vendor/roave/backward-compatibility-check/src/Command/AssertBackwardsCompatible.php(226): Roave\BackwardCompatibility\Git\PickLastMinorVersionFromCollection->forVersions(Object(Version\VersionsCollection))
#1 /repo/vendor/roave/backward-compatibility-check/src/Command/AssertBackwardsCompatible.php(150): Roave\BackwardCompatibility\Command\AssertBackwardsCompatible->determineFromRevisionFromRepository(Object(Roave\BackwardCompatibility\Git\CheckedOutRepository), Object(Symfony\Component\Console\Output\StreamOutput))
#2 /repo/vendor/symfony/console/Command/Command.php(255): Roave\BackwardCompatibility\Command\AssertBackwardsCompatible->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Con in /repo/vendor/roave/backward-compatibility-check/src/Git/PickLastMinorVersionFromCollection.php on line 48
```

But we should have "enough" stable tags to compare...

```
root@f5eb6f0ecb4d:/repo# git tag
0.1.0
0.2.0
1.0.0
1.1.0
1.1.1
2.0.0
2.0.0-alpha1
2.0.0-beta1
2.1.0
2.1.1
2.1.2
2.1.3
2.2.0
2.2.1
2.2.2
2.2.3
2.2.4
2.3.0
2.3.1
2.3.2
2.3.3
2.3.4
2.4.0
2.4.0-beta1
2.4.0-rc1
2.4.1
2.4.2
2.4.3
2.4.4
2.5.0
2.5.0-rc1
2.6.0
2.6.1
2.6.2
2.7.0-beta1
3.0.0-alpha1
```

### PR DESCRIPTION
**Pull request by Ocramius**
Fixes #114 
Fixes #117

### CODE REVIEW

File: test/unit/Git/PickLastMinorVersionFromCollectionTest.php [hunk 1]
```diff
@@ -28,6 +28,7 @@ public function lastStableMinorVersionForCollectionProvider() : array
             ['2.2.0', ['1.2.0', '2.2.0', '2.2.1', '1.2.1']],
             ['2.2.0', ['1.2.0', '2.2.0', '2.2.0-alpha1', '2.2.1', '1.2.1']],
             ['2.2.0', ['1.2.0', '2.2.0-alpha1', '2.2.0', '2.2.1', '1.2.1']],
+            ['2.6.0', ['1.1.1', '3.0.0-alpha1', '2.7.0-beta1', '2.6.2', '2.0.0', '2.6.1', '2.6.0']],
```
Comment (schmunk42) [inline test/unit/Git/PickLastMinorVersionFromCollectionTest.php]: Shouldn't this be `2.6.2`?

Comment (Ocramius) [reply test/unit/Git/PickLastMinorVersionFromCollectionTest.php]: I've explicitly changed it in https://github.com/Roave/BackwardCompatibilityCheck/pull/118/commits/33e851fd00086a77f053e3d80132402dd31e8bd7 (see commit message)

Comment (schmunk42) [reply test/unit/Git/PickLastMinorVersionFromCollectionTest.php]: Sorry, didn't get the "lastMinorVersion" thingy....

Comment (Ocramius) [reply test/unit/Git/PickLastMinorVersionFromCollectionTest.php]: That's alright - it was a design decision :-)

**Pull request merged**